test/fuzz/test_runner.py

```diff
@@ -259,9 +260,16 @@ def job(command, t, t_env):
     for target, t_env in targets:
         target_corpus_dir = corpus_dir / target
         os.makedirs(target_corpus_dir, exist_ok=True)
+        use_value_profile = int(random.random() < .3)
+        max_len = 0 if random.random() < .7 else 2**random.randint(6, 12)
+        mutate_depth = random.randint(1, 10)
         command = [
             os.path.join(build_dir, 'src', 'test', 'fuzz', 'fuzz'),
-            "-runs=100000",
+            "-max_total_time=6000",
```

Comment:
That's a large increase in runtime isn't it? Won't 100k iterations take much less than 100 minutes to complete on most of our targets?

We've got 115 targets at the moments, this change makes it so running this script with `--generate` for all targets would take around 8 days.

Reply:
Ok i'm confused. I'm aware i just gave this a superficial look but i'm absolutely not following here. *(Last comment and if i don't get it i'll look into it more in depth.)*

> The number of tasks that are spawned is ~300

How comes? The default for `--par` is `4`.

> So even if you run each task only for a minute, it will take longer than 4 hours already

How could, with 300 tasks, running 169 tasks for 1 minute take 4 hours?

----

My main worry would be that increasing the runtime of the script by, i don't know, 100X, would make it so people would always stop it before it completes thereby never generating seeds for the last (alphabetically-sorted) targets.

I may be missing underestimating how many tasks people usually spawn when running this script though. But `--par 169` sounds like a lot.

Reply:
Indeed a too short runtime could make the script useless (since people would basically start running it in a loop itself), but a too long one would likely introduce a bias toward the first (alphabetically sorted) targets.

I guess it would be helpful to know who uses this script and how. For instance if they are continuously generating coverage most likely they are already running it in a loop of some sort and therefore a runtime of like 10minutes per target would spead the time spent over all targets instead of generating more for the first ones.

But i'll stop bikeshedding now, your change looks good. If someone wants to change it they can open a PR and we'll learn more about how the script is being used.

Reply:
I tried using this script, but got away from that because it always tries to run all fuzz targets. I would either only be able to fuzz each target very briefly of which a big portion would just be the initialization, or if I were to kill the fuzzing when I need my computer for other things, always fuzz the same targets due to its determinism. I recently ended up making myself a bash script instead that randomly picks ten fuzz targets and fuzzes them for one hour each (with 12 processes in parallel). I run this nightly per a cronjob:

`for i in $(find ../qa-assets-active-fuzzing/fuzz_seed_corpus/ -mindepth 1 -maxdepth 1 -type d | shuf | head -n10); do FUZZ=$(basename $i) src/test/fuzz/fuzz -jobs=12 -reload=1 -max_total_time=3600 $i; done`